code/obj/machinery/computer/operating.dm

```diff
@@ -26,56 +28,262 @@
+		return
 	if(src.table && (src.table.check_victim()))
 		src.victim = src.table.victim
-		dat += {"
-<B>Patient Information:</B><BR>
-<BR>
-<B>Name:</B> [src.victim.real_name]<BR>
-<B>Age:</B> [src.victim.bioHolder.age]<BR>
-<B>Blood Type:</B> [src.victim.bioHolder.bloodType]<BR>
-<BR>
-<B>Health:</B> [src.victim.health]<BR>
-<B>Brute Damage:</B> [src.victim.get_brute_damage()]<BR>
-<B>Toxins Damage:</B> [src.victim.get_toxin_damage()]<BR>
-<B>Fire Damage:</B> [src.victim.get_burn_damage()]<BR>
-<B>Suffocation Damage:</B> [src.victim.get_oxygen_deprivation()]<BR>
-<B>Patient Status:</B> [src.victim.stat ? "Non-responsive" : "Stable"]<BR>
-"}
 	else
 		src.victim = null
-		dat += {"
-<B>Patient Information:</B><BR>
-<BR>
-<B>No Patient Detected</B>
-"}
-	user.Browse(dat, "window=op")
-	onclose(user, "op")
-
-/obj/machinery/computer/operating/Topic(href, href_list)
-	if(..())
-		return
-	if ((usr.contents.Find(src) || (in_interact_range(src, usr) && istype(src.loc, /turf))) || (issilicon(usr)))
-		src.add_dialog(usr)
-//		if (href_list["update"])
-//			src.interacted(usr)
-	return
+		src.gene_scan = null
+		src.victim_data = null
+	if (src.victim)
+		src.gene_scan = create_new_dna_sample_file(src.victim)
+		src.victim_data += list(sample_victim())
+		if (length(src.victim_data) > src.history_max)
+			src.victim_data.Cut(1, 2) //drop the oldest entry
+		use_power(500)
 
-/obj/machinery/computer/operating/process()
-	..()
-	if (status & (BROKEN | NOPOWER))
+
+/obj/machinery/computer/operating/proc/sample_victim()
+	. = list()
+	.["brute"] = src.victim?.get_brute_damage()
+	.["burn"] = src.victim?.get_burn_damage()
+	.["toxin"] = src.victim?.get_toxin_damage()
+	.["oxygen"] = src.victim?.get_oxygen_deprivation()
+
+/obj/machinery/computer/operating/ui_data(mob/user)
+	. = list()
+	.["occupied"] = istype(src.victim)
+	if(!src.victim)
 		return
-	use_power(250)
+	var/datum/color/blood_color_value = new()
+	blood_color_value.from_hex(src.victim.blood_color)
+
+	// hack for hemoglyph b/c we don't tint blood color properly
+	if(src.victim.bioHolder.GetEffect("roach"))
+		blood_color_value.from_hex("#009E81") // commented out blood color in `mutantraces.dm`. yep.
+
+	var/datum/statusEffect/simpledot/radiation/R = src.victim.hasStatus("radiation")
+	if (R?.stage)
+		.["rad_stage"] = R.stage
+		.["rad_dose"] = src.victim.radiation_dose
+	else
+		.["rad_stage"] = 0
+		.["rad_dose"] = 0
+
+	.["patient_name"] = src.victim.real_name
+	.["victim_status"] = src.victim.stat
+
+	.["body_temp"] = src.victim.bodytemperature
+	.["optimal_temp"] = src.victim.base_body_temp
+
+	.["victim_data"] = src.victim_data
+
+	.["max_health"] = round(src.victim.max_health)
+	.["health"] = round(src.victim.health)
+	.["brute"] = round(src.victim.get_brute_damage())
+	.["burn"] = round(src.victim.get_burn_damage())
+	.["toxin"] = round(src.victim.get_toxin_damage())
+	.["oxygen"] = round(src.victim.get_oxygen_deprivation())
+
+	.["blood_pressure"] = src.victim.blood_pressure
```

Comment:
maybe group these up into a sub-list here on the data side, and have the component expand it out?